sqlglot/dialects/starrocks.py

```diff
@@ -81,4 +164,27 @@ class Generator(MySQL.Generator):
             exp.UnixToTime: rename_func("FROM_UNIXTIME"),
         }
 
+        def create_sql(self, e: exp.Create) -> str:
+            # Starrocks's primary is defined outside the schema, so we need to move it into after engine properties
+            schema = e.this
+            if isinstance(schema, exp.Schema):
+                primary_key = next(
+                    (expr for expr in schema.expressions if isinstance(expr, exp.PrimaryKey)),
+                    None,
+                )
+                if primary_key:
+                    schema.expressions.remove(primary_key)
+                    props = e.args.get("properties")
+                    if props:
+                        # Verify if the first one is an engine property. Is true then insert it after the engine,
+                        # otherwise insert it at the beginning
+                        if isinstance(props.expressions[0], exp.EngineProperty):
+                            props.expressions.insert(1, primary_key)
```

Comment:
This is important in starrocks, otherwise if the order is wrong, we will get the following error:
![image](https://github.com/user-attachments/assets/a4142e86-5e3d-4232-8116-b702adce1392)

We can notice in create syntax that [key_desc] should be after [ENGINE type], unless ENGINE type is not written, in which case key_desc should be first.
https://docs.starrocks.io/docs/sql-reference/sql-statements/data-definition/CREATE_TABLE/#syntax